### DISCUSSION TITLE
Grapql viewer repositories gets only repositories where the user is collaborator

### DISCUSSION THREAD

**Discussion by ajanoni** [community/community#24861] [Apps, API and Webhooks]
<p>I am trying to run this query with my account by it is only returning repositories which I am contributor:</p>
<pre><code>query ($first: Int, $after: String, $last: Int, $before: String, $sortDirection: OrderDirection!) {
  viewer {
    repositories(affiliations: [OWNER, ORGANIZATION_MEMBER, COLLABORATOR], first: $first, after: $after, last: $last, before: $before, orderBy: {field: NAME, direction: $sortDirection}, isFork: false) {
      totalCount
      nodes {
        name
        url
        updatedAt
        isFork
        primaryLanguage {
          name
        }
        defaultBranchRef {
          name
          target {
            ... on Commit {
              history(first: 1) {
                nodes {
                  authoredDate
                }
              }
            }
          }
        }
        owner {
          login
        }
        languages(first: 100) {
          edges {
            size
            node {
              name
            }
          }
        }
      }
      pageInfo {
        startCursor
        endCursor
        hasNextPage
        hasPreviousPage
      }
    }
  }
}

{
"sortDirection": "ASC",
"first": 100
}
</code></pre>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/ajanoni">@ajanoni</a>,</p>
<p>Thank you for being here! Could you try to enable “Organization access” when authorizing the GraphQL API Explorer?</p>

**Comment by ajanoni** [comment]
<p>I am using a personal access token with all grants enabled. </p>

**Comment by ajanoni** [comment]
<p>And when I use </p>
<pre><code>query ($first: Int, $after: String, $last: Int, $before: String, $sortDirection: OrderDirection!) {
organization(name: myorgname {
repositories
</code></pre>
<p>Then it returns all organization repositories. So I guess it is not an access grant issue.</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hey <a class="mention" href="/ajanoni">@ajanoni</a>, </p>
<p>Thanks for the update! I think you might’ve stumbled on a bug here…I’ll report it to the API team, and will post an update here once I hear back.</p>
<p>Thanks again!</p>
<p>Andrea</p>

**Comment by krasimir** [comment]
<p><a class="mention" href="/ajanoni">@ajanoni</a> do you have any news on this one. I’m also in a need to fetch all the repos associated with my account but I’m only getting the public repos. Similar request works just fine with the v3 API.</p>

**Comment by krasimir** [comment]
<p>I can confirm that this is probably a bug because I do see private repos using graphql API with the same public token when I’m doing a search. For example:</p>
<pre><code>{
  query: {
      search(query: "org:&lt;my company name here&gt;", type: REPOSITORY, first: 100) {
          repositoryCount
          edges {
              cursor,
              node {
                  ... on Repository {
                      name,
                    id,
                    nameWithOwner,
                    homepageUrl
                  }
              }
          }
      }
  }
}
</code></pre>
<p>shows public repos and private repos owned by <code>\&lt;my company name here\&gt;</code>. Still not perfect because it is not showing the forks but still.</p>

**Comment by vachi** [comment]
<p>Still hitting this bug. </p>
<p>Is the solution using the rest API for grabbing repos?</p>

**Comment by mdelagrange** [comment]
<blockquote>
<p><a class="mention" href="/vachi">@vachi</a> wrote:</p>
</blockquote>
<blockquote>
<p>Still hitting this bug. </p>
</blockquote>
<blockquote>
<p>Is the solution using the rest API for grabbing repos?</p>
</blockquote>
<p>Hi <a class="mention" href="/vachi">@vachi</a>!</p>
<p>Hopefully, you found a workaround for this issue, but you may want to try combinations of the affiliations and ownerAffiliations arguments (possibly both!).  I ran into this issue in GitHub Enterprise Server, and these were the query arguments that worked for me:</p>
<pre><code>query AllRepositories($cursor: String) {
  viewer {
    repositories(first:100, ownerAffiliations: [ORGANIZATION_MEMBER], affiliations: [ORGANIZATION_MEMBER], isFork: false, after: $cursor) {
      edges {
        node {
          name
          nameWithOwner
        }
      }
    }
  }
}
</code></pre>
<p>This started happening when <span>ownerAffiliations</span> was added to GitHub Enterprise Server 2.16 and to <a href="http://github.com" rel="nofollow noopener">github.com</a>.  When I contacted Enterprise support to debug the new query behavior, this was their response:</p>
<p>I believe the behaviour you’ve reported is as a result of a change that shipped in GitHub Enterprise 2.16 release series, where an<span> </span>ownerAffiliations<span> </span>parameter was added to the<span> </span><a href="https://developer.github.com/v4/object/repository/" target="_blank" rel="nofollow noopener"><em>Repositories</em><span> </span>connection</a><span> </span>in our GraphQL API.</p>
<p>Before this change, the<span> </span>affiliations<span> </span>field was overloaded to mean both the affiliations that the<span> </span><em>viewer</em><span> </span>had with the specified repositories, as well as the affiliations between the<span> </span><em>user</em><span> </span>that the connection was operating on. This caused a host of issues in various places, and we opted to separate them out to allow for more flexibility.</p>
<p>In order to fix the above, you need to pass the relevant list of arguments to both connections, which should provide you all of the possible repositories. That is, for your query, pass both the<span> </span>ownerAffiliations: [ORGANIZATION_MEMBER]<span> </span>and<span> </span>affiliations: [ORGANIZATION_MEMBER]<span> </span>repository affiliation arguments.</p>
<ul>
<li>Morgan</li>
</ul>

**Comment by insidewhy** [comment]
<p>This doesn’t work for me.</p>

**Comment by insidewhy** [comment]
<p>Hi Andrea, has the team gotten back to you yet? This bug has existed for quite some time now and the only solution seems to be to go back to using the v3 API.</p>

**Comment by insidewhy** [comment]
<p>Ah but it lead me to the correct solution: <code>affiliations:[OWNER, ORGANIZATION_MEMBER, COLLABORATOR],&amp;nbsp;ownerAffiliations:[OWNER, ORGANIZATION_MEMBER, COLLABORATOR],</code> gets me the private repositories! Thank you!</p>

**Comment by insidewhy** [comment, marked as answer]
<p>Never mind Andrea, I found the solution: need to set <code>affiliations:[OWNER, ORGANIZATION_MEMBER, COLLABORATOR],&amp;nbsp;ownerAffiliations:[OWNER, ORGANIZATION_MEMBER, COLLABORATOR]</code>. Not sure why this is necessary for private organisations and not public organisations, but at least I can get all the repos I have access to now!</p>
